Improvements in segment plotting

chellings opened the pull request:
Various improvements in segment plotting:
- sort plots by instrument names (or by order in instruments if instruments is provided)
- sort lines in plots by channel names
- add instrument/qubit name as subplot title
- add AWG name to legend entry when plotting per qubit
- corrected position of figure title 

Tested on a virtual setup (and partially on XLD).

FYI @antsr

chellings committed:
Segment.plot: add instrument/qubit name as subplot title

chellings committed:
segment.plot: sort plots by instrument names (or by order in instruments if instruments is provided)

chellings committed:
segment.plot: sort lines in plots by channel names

chellings committed:
segment.plot: add AWG name to legend entry when plotting per qubit

chellings committed:
segment.plot: corrected position of figure title

stephlazar commented on pycqed/measurement/waveform_control/segment.py:
```diff
@@ -1070,7 +1081,7 @@ def plot(self, instruments=None, channels=None, legend=True,
+            fig.suptitle(f'{self.name}', y=1.01)
             plt.tight_layout()
             if savefig:
                 plt.savefig(f'{self.name}.png')
```
I think at some point I added a plt.close(fig). Is this not needed?

chellings replied:
I guess this was removed when the function was changed by Nathan to not show the plots, but return them instead. If we close them here, they cannot be shown afterwards. Since they are returned, the caller is responsible for closing them (which we probably don't do, when calling it from a notebook.) Not sure, what behavior we want. Producing screen output inside a function is not desirable either, that is why Nathan remove the .show(). However, we could add parameters closefig and plotfig, and issue a warning if closefig and not savefig and not plotfig. What do you think?

chellings replied:
Ah, no, it was apparently not removed: you have changed it on the parallel 2qb calib branch, that is why it is not in the master. But then you return a closed figure, which does not really help.

stephlazar replied:
I think we should have the flag close_fig=True by default. And if close_fig then it calls plt.show() and does not return figure, else it does not call plt.show() + returns figure.

Commit:
Aesthetic fixes to the code in Segment.plot()

Commit:
Merge branch 'qudev_master' into enh/segment_plot

chellings committed:
Merge branch 'qudev_master' into enh/segment_plot

chellings committed:
Segment/Sequence: added optional bool param show_and_close to plot()
If True (default) figures will be shown and closed, if False, they will be returned instead.

stephlazar reviewed (approved):
Looks good!
